spec/openapi.yaml

```diff
@@ -53,182 +156,498 @@ paths:
+    Category:
       type: object
       properties:
         apiVersion:
           type: string
         data:
-          $ref: '#/components/schemas/ApiKeyResponse_data'
+          $ref: '#/components/schemas/Category_data'
+        has_next:
+          type: integer
+        has_previous:
+          type: integer
+        number_of_pages:
+          type: integer
+        page:
+          type: integer
+        records_per_page:
+          type: integer
         status:
           type: string
-    inline_response_200_data:
+        status_code:
+          type: integer
+        total_count:
+          type: integer
+    Category_data:
+      type: array
+      items:
+        type: object
+        properties:
+          id:
+            type: integer
+          name:
+            type: string
+
+    Language:
       type: object
       properties:
-        name:
+        apiVersion:
           type: string
-        id:
+        data:
+          $ref: '#/components/schemas/Language_data'
+        has_next:
+          type: integer
+        has_previous:
+          type: integer
+        number_of_pages:
+          type: integer
+        page:
           type: integer
-    inline_response_200:
-      type: object
-      properties:
         records_per_page:
           type: integer
-        apiVersion:
+        status:
           type: string
         status_code:
           type: integer
+        total_count:
+          type: integer
+    Language_data:
+      type: object
+      properties:
+        id:
+          type: integer
+        name:
+          type: string
+    
+    SearchResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
         data:
           type: array
           items:
-            $ref: '#/components/schemas/inline_response_200_data'
+            type: object
+            $ref: '#/components/schemas/Resource_data'
         number_of_pages:
           type: integer
+        page:
+          type: integer
+        records_per_page:
+          type: integer
+        status:
+          type: string
+        status_code:
+          type: integer
         total_count:
           type: integer
+    FilteredResourcesResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+        data:
+          type: array
+          items:
+            type: object
+            $ref: '#/components/schemas/Resource_data'
         has_next:
-          type: boolean
-        has_prev:
-          type: boolean
+          type: integer
+        has_previous:
+          type: integer
+        number_of_pages:
+          type: integer
         page:
           type: integer
+        records_per_page:
+          type: integer
         status:
           type: string
+        status_code:
+          type: integer
+        total_count:
+          type: integer
+    ApiKeyResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
```

Comment:
I think it would be helpful to have descriptions here on all of these. Is there a way to DRY up this yaml file by having a central location for things that are repeated such as `apiVersion` so that way we can list the type and description once but ensure that it shows up in the responses and examples for each of the endpoints?

If not, that's fine but it will be a super big pain to change something and remember to change it everywhere and ensure that everything is correct.

Reply:
I totally agree and was thinking I would circle back and refactor the components section to see if we could get more $ref type entries in the endpoint sections and then the responses and examples defined in the components.